google/cloud/storage/client.h

```diff
@@ -2389,11 +2439,23 @@ class Client {
                           Options&&... options) {
     internal::DeleteNotificationRequest request(bucket_name, notification_id);
     request.set_multiple_options(std::forward<Options>(options)...);
-    raw_client_->DeleteNotification(request);
+    ThrowOnError(raw_client_->DeleteNotification(request));
   }
   //@}
 
  private:
+  template <typename T>
+  StatusOr<T> AsStatusOr(std::pair<Status, T> result) {
+    if (not result.first.ok()) {
+      return std::move(result.first);
+    }
+    return std::move(result.second);
+  }
+
+  Status AsStatusOr(std::pair<Status, internal::EmptyResponse> result) {
```

Comment:
"AsStatusOr" yet returns a Status?

Reply:
Fixed.  We have to make our minds about `StatusOr<void>` vs. `Status`, I do not have a super strong opinion one way or the other, but this PR is not when we decide that, and `Status` is more natural.

Reply:
Which is to say that I was using  `StatusOr<void>` then changed my mind but did not change the function, thanks for pointing this out.